src/MessageReader.js

```diff
@@ -39,22 +39,33 @@ class StandardTypeReader {
     this.view = new DataView(buffer.buffer, buffer.byteOffset);
   }
 
+  json(): { [key: string]: any } {
+    const resultString = this.string();
+    try {
+      return JSON.parse(resultString);
+    } catch {
+      return { err: `Could not parse ${resultString}` };
+    }
+  }
+
   string() {
     const len = this.int32();
     const codePoints = new Uint8Array(this.buffer.buffer, this.buffer.byteOffset + this.offset, len);
     this.offset += len;
     // if the string is relatively short we can use apply
     // but very long strings can cause a stack overflow due to too many arguments
-    // in those cases revert to a slower itterative string building approach
+    // in those cases revert to a slower iterative string building approach
+    let resultString = "";
     if (codePoints.length < 1000) {
-      return String.fromCharCode.apply(null, codePoints);
+      resultString = String.fromCharCode.apply(null, codePoints);
     }
 
-    let data = "";
-    for (let i = 0; i < len; i++) {
-      data += String.fromCharCode(codePoints[i]);
+    if (!resultString.length) {
+      for (let i = 0; i < len; i++) {
+        resultString += String.fromCharCode(codePoints[i]);
+      }
     }
-    return data;
+    return resultString;
```

Comment:
Let's maybe revert this back to the code before, since we didn't really have to touch this after all (except for the typo in the comment)?

Reply:
Reverted.